java.net.UnknownHostException is thrown if the hostname cannot successfully determine the systems IP

Stacktrace:

```
$ java -jar testing-0.1.0-SNAPSHOT-standalone.jar 
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Exception in thread "main" java.net.UnknownHostException: ip-10-0-1-209: ip-10-0-1-209: Name or service not known
    at java.net.InetAddress.getLocalHost(InetAddress.java:1496)
    at riemann.codec$encode_client_pb_event.invoke(codec.clj:95)
    at riemann.client$send_event.invoke(client.clj:72)
    at testing.core$_main.invoke(core.clj:12)
    at clojure.lang.AFn.applyToHelper(AFn.java:152)
    at clojure.lang.AFn.applyTo(AFn.java:144)
    at testing.core.main(Unknown Source)
Caused by: java.net.UnknownHostException: ip-10-0-1-209: Name or service not known
    at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
    at java.net.InetAddress$1.lookupAllHostAddr(InetAddress.java:922)
    at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1316)
    at java.net.InetAddress.getLocalHost(InetAddress.java:1492)
    ... 6 more
```

If I add `ip-10-0-1-209` to `/etc/hosts` the issue goes away; I'm assuming this is because of a reverse lookup failure for the host and the fact that AWS doesn't add the hostname to `/etc/hosts`.

Sample code:

```
(ns testing.core
  (:gen-class))

(defn -main 
    (require '[riemann.client :as r])
    (def c (r/tcp-client {:host "127.0.0.1"}))
    (-> c (r/send-event {:service "foo" :state "ok"})
        (deref 5000 ::timeout))
    @(r/query c "state = \"ok\"")
    )
```

--- a/src/riemann/codec.clj
+++ b/src/riemann/codec.clj
@@ -12,17 +12,6 @@
 
 (def event-keys (set (map keyword (Event/getBasis))))
 
-(defn assoc-default
-  "Like assoc, but only alters the map if it does not already contain the given
-  key.
-
-  (assoc-default {} :foo 2)         ; {:foo 2}
-  (assoc-default {:foo nil} :foo 2) ; {:foo nil}"
-  [m k v]
-  (if (contains? m k)
-    m
-    (assoc m k v)))
-
 (defn decode-pb-query
   "Transforms a java protobuf Query to a Query."
   [^Proto$Query q]
@@ -91,10 +80,11 @@
   "Clients usually fill in an event's local host and time automatically, if
   not explicitly specified."
   [e]
-  (-> e
-    (assoc-default :time (/ (System/currentTimeMillis) 1000))
-    (assoc-default :host (.. InetAddress getLocalHost getHostName))
-    encode-pb-event))
+  (let [absent? (-> e keys set complement)]
+    (encode-pb-event
+     (cond-> e
+       (absent? :host) (assoc :host (.. InetAddress getLocalHost getHostName))
+       (absent? :time) (assoc :time (/ (System/currentTimeMillis) 1000))))))
 
 (defn decode-pb-msg
   "Transforms a java protobuf Msg to a defrecord Msg."

